pkg/web/apps/apps_http_handler.go

```diff
@@ -45,28 +47,64 @@ func (a *httpHandler) GetApps(c echo.Context) error {
 	return c.JSON(http.StatusOK, apps)
 }
 
-// GetAppByID returns an app with the ID in JSON format from the AppService
-// TODO: Implement
 //GetAppById returns a app with the ID in JSON format from the AppService
 func (a *httpHandler) GetAppByID(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+	app, err := a.Service.GetAppByID(c.Param("id"))
 
 	if err != nil {
 		return err
 	}
 
-	return c.JSON(http.StatusOK, apps)
+	return c.JSON(http.StatusOK, app)
 }
 
-// UpdateApp returns an app updated with the ID in JSON format from the AppService
-// TODO: Implement
 //UpdateApp returns a app updated with the ID in JSON format from the AppService
-func (a *httpHandler) UpdateApp(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+func (a *httpHandler) UpdateAppVersions(c echo.Context) error {
+	// Validations
+	versions := []models.Version{}
+	errV := json.NewDecoder(c.Request().Body).Decode(&versions)
+
+	// check if the data sent is in the correct format
+	if errV != nil {
+		return httperrors.BadRequest(c, "Invalid data")
+	}
+
+	// Check if versions were sent all the body is empty
+	if len(versions) == 0 {
+		return httperrors.BadRequest(c, "No version(s) was sent.")
+	}
+
+	errUpdate := a.Service.UpdateAppVersions(versions)
+
+	if errUpdate != nil {
+		return httperrors.GetHTTPResponseFromErr(c, errUpdate)
+	}
+
+	return c.JSON(http.StatusOK, "")
+}
+
+//UpdateApp returns a app updated with the ID in JSON format from the AppService
+func (a *httpHandler) UpdateAllAppVersions(c echo.Context) error {
+	versionRes := models.VersionRes{}
```

Comment:
Is it according to your ideas? If not, how would like that it be implemented?

Reply:
With my suggestion in the router. You can reuse the `version` model.